File: Merge In Between Linked Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
    b += 2 - a

    pt := list1
    for a > 1 {
        pt = pt.Next
        a--
    }

    pt2 := pt
    for b > 0 {
        pt2 = pt2.Next
        b--
    }

    pt.Next = list2
    for pt.Next != nil {
        pt = pt.Next
    }

    pt.Next = pt2
    return list1
}

File: Merge In Between Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        b += 2 - a;
        
        auto pt = list1;
        while (a > 1) {
            pt = pt->next;
            a--;
        }

        auto pt2 = pt;
        while (b > 0) {
            pt2 = pt2->next;
            b--;
        }

        pt->next = list2;
        while (pt->next) pt = pt->next;
        pt->next = pt2;
        return list1;
    }
};
